pychunkedgraph/meshing/manifest.py

```diff
@@ -262,29 +273,44 @@ def _get_mesh_paths(cg, node_ids: Sequence[np.uint64], stop_layer: int = 2,) ->
     return result
 
 
-def _get_children_before_start_layer(cg, node_id: np.uint64, start_layer: int = 6):
+def _get_children_before_start_layer(
+    cg, node_id: np.uint64, start_layer: int = 6, bounding_box=None
+):
     if cg.get_chunk_layer(node_id) == 2:
         return np.array([node_id], dtype=NODE_ID)
     result = [empty_1d]
     parents = np.array([node_id], dtype=np.uint64)
     while parents.size:
         children = cg.get_children(parents, flatten=True)
+        bound_mask = misc_utils.mask_nodes_by_bounding_box(
+            cg.meta, children, bounding_box=bounding_box
+        )
         layers = cg.get_chunk_layers(children)
-        result.append(children[layers <= start_layer])
-        parents = children[layers > start_layer]
+        result.append(children[(layers <= start_layer) & bound_mask])
+        parents = children[(layers > start_layer) & bound_mask]
     return np.concatenate(result)
 
 
 def children_meshes_sharded(
-    cg, node_id: np.uint64, stop_layer=2, verify_existence=False, bounding_box=None,
+    cg,
+    node_id: np.uint64,
+    stop_layer=2,
+    start_layer=None,
+    bounding_box=None,
+    verify_existence=False,
 ):
     """
     For each ID, first check for new meshes,
     If not found check initial meshes.
     """
-    MAX_STITCH_LAYER = cg.meta.custom_data.get("mesh", {}).get("max_layer", 2)
+    if start_layer is None:
```

Comment:
I think else is not needed if done this way
```
    MAX_STITCH_LAYER = cg.meta.custom_data.get("mesh", {}).get("max_layer", 2)
    if start_layer is not None:
        MAX_STITCH_LAYER = start_layer
```

Reply:
fixed!